### PR TITLE
SBERDOMA-496 Added /contact/create page

### PR DESCRIPTION
**Pull request by SavelevMatthew**
Added form for contact creation. 
Need to test corner cases of form usages and then we are ready to pull 👍

### CODE REVIEW

File: apps/condo/domains/contact/components/CreateContactForm.tsx [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+
+export const CreateContactForm: React.FC = () => {
+    const intl = useIntl()
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    const { organization } = useOrganization()
+    const router = useRouter()
+
+    const FullNameLabel = intl.formatMessage({ id: 'field.FullName.short' })
```
Comment (pahaz) [inline apps/condo/domains/contact/components/CreateContactForm.tsx:44]: translations should be at the top of the component

File: apps/condo/domains/contact/components/CreateContactForm.tsx [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+    const FullNamePlaceholder = intl.formatMessage({ id:'field.FullName' })
+    const PhoneLabel = intl.formatMessage({ id: 'Phone' })
+    const FieldIsRequiredMsg = intl.formatMessage({ id: 'FieldIsRequired' })
+    const PhoneIsNotValidMsg = intl.formatMessage({ id: 'pages.auth.PhoneIsNotValid' })
+    const ExamplePhoneMsg = intl.formatMessage({ id: 'example.Phone' })
+    const EmailLabel = intl.formatMessage({ id: 'field.EMail' })
+    const EmailError = intl.formatMessage({ id: 'pages.auth.EmailIsNotValid' })
```
Comment (pahaz) [inline apps/condo/domains/contact/components/CreateContactForm.tsx:51]: translation const should ends with Label/Message suffix

File: apps/condo/domains/contact/components/CreateContactForm.tsx [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+        email: [
+            {
+                type: 'email',
+                message: EmailError,
+            },
+        ],
+        property: [
+            {
+                required: true,
+                message: intl.formatMessage({ id: 'field.Property.requiredError' }),
```
Comment (pahaz) [inline apps/condo/domains/contact/components/CreateContactForm.tsx:80]: all translations should be at the top

File: apps/condo/domains/contact/components/CreateContactForm.tsx [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+            },
+        ],
+        unit: [
+            {
+                required: true,
+                message: intl.formatMessage({ id: 'field.Unit.requiredError' }),
```
Comment (pahaz) [inline apps/condo/domains/contact/components/CreateContactForm.tsx:86]: extract the const

File: apps/condo/domains/contact/components/CreateContactForm.tsx [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+            },
+        ],
+        name: [
+            {
+                required: true,
+                message: intl.formatMessage({ id: 'field.FullName.requiredError' }),
```
Comment (pahaz) [inline apps/condo/domains/contact/components/CreateContactForm.tsx:92]: extract

File: apps/condo/domains/contact/components/ErrorsContainer.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import styled from '@emotion/styled'
+import { useIntl } from '@core/next/intl'
+import React from 'react'
+import { colors } from '@condo/domains/common/constants/style'
+
+export const ErrorsWrapper = styled.div`
```
Comment (pahaz) [inline apps/condo/domains/contact/components/ErrorsContainer.tsx:6]: why do you need to export it?

File: apps/condo/domains/contact/components/CreateContactForm.tsx (hunk 1 above)
Comment (SavelevMatthew) [reply apps/condo/domains/contact/components/CreateContactForm.tsx:38]: done 👍

File: apps/condo/domains/contact/components/CreateContactForm.tsx (hunk 2 above)
Comment (SavelevMatthew) [reply apps/condo/domains/contact/components/CreateContactForm.tsx]: done 👍

File: apps/condo/domains/contact/components/CreateContactForm.tsx (hunk 3 above)
Comment (SavelevMatthew) [reply apps/condo/domains/contact/components/CreateContactForm.tsx]: done 👍

File: apps/condo/domains/contact/components/CreateContactForm.tsx (hunk 4 above)
Comment (SavelevMatthew) [reply apps/condo/domains/contact/components/CreateContactForm.tsx]: done 👍

File: apps/condo/domains/contact/components/CreateContactForm.tsx (hunk 5 above)
Comment (SavelevMatthew) [reply apps/condo/domains/contact/components/CreateContactForm.tsx]: done 👍

File: apps/condo/domains/contact/components/ErrorsContainer.tsx (hunk 6 above)
Comment (SavelevMatthew) [reply apps/condo/domains/contact/components/ErrorsContainer.tsx]: done 👍

**Review by dkoviazin — APPROVED** [review]
Ok

**Review by toplenboren — APPROVED** [review]

File: apps/condo/domains/contact/components/CreateContactForm.tsx [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+        name: [
+            {
+                required: true,
+                message: FullNameRequiredMessage,
+            },
+        ],
+    }
+
+    const [selectedPropertyId, setSelectedPropertyId] = useState(null)
+    const selectPropertyIdRef = useRef(selectedPropertyId)
```
Comment (AntonAL) [inline apps/condo/domains/contact/components/CreateContactForm.tsx:98]:
```suggestion
    const selectedPropertyIdRef = useRef(selectedPropertyId)
```

File: apps/condo/domains/contact/components/CreateContactForm.tsx [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+        router.push('/contact/')
+    })
+    return (
+        <FormWithAction
+            action={action}
+            layout={'horizontal'}
+            validateTrigger={['onBlur', 'onSubmit']}
+            colon={false}
+            formValuesToMutationDataPreprocessor={(values) => {
+                values.property = selectPropertyIdRef.current
```
Comment (AntonAL) [inline apps/condo/domains/contact/components/CreateContactForm.tsx:116]:
```suggestion
                values.property = selectedPropertyIdRef.current
```

File: apps/condo/domains/contact/components/CreateContactForm.tsx [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+                required: true,
+                message: FullNameRequiredMessage,
+            },
+        ],
+    }
+
+    const [selectedPropertyId, setSelectedPropertyId] = useState(null)
+    const selectPropertyIdRef = useRef(selectedPropertyId)
+    useEffect(() => {
+        selectPropertyIdRef.current = selectedPropertyId
```
Comment (AntonAL) [inline apps/condo/domains/contact/components/CreateContactForm.tsx:100]:
```suggestion
        selectedPropertyIdRef.current = selectedPropertyId
```

File: apps/condo/pages/contact/create.tsx [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+import React from 'react'
+import { CreateContactForm } from '@condo/domains/contact/components/CreateContactForm'
+
+interface IPageWithHeaderAction extends React.FC {
+    headerAction?: JSX.Element
+}
+
+const CreateContactPage: IPageWithHeaderAction = () => {
+    const intl = useIntl()
+    const PageTitle = intl.formatMessage({ id: 'contact.AddContact' })
```
Comment (AntonAL) [inline apps/condo/pages/contact/create.tsx:18]: Please, add appropriate translation into json-files.

File: apps/condo/domains/contact/components/CreateContactForm.tsx [hunk 11]
```diff
@@ -0,0 +1,246 @@
...
+const BottomLineWrapper = styled.div`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  justify-content: flex-start;
+`
+
+export const CreateContactForm: React.FC = () => {
+    const intl = useIntl()
+    const FullNameLabel = intl.formatMessage({ id: 'field.FullName.short' })
```
Comment (AntonAL) [inline apps/condo/domains/contact/components/CreateContactForm.tsx:38]: Missing translation for this key.
Please, add it into json-files

**Review by AntonAL — CHANGES_REQUESTED** [review]

File: apps/condo/domains/contact/components/CreateContactForm.tsx (hunk 11 above)
Comment (SavelevMatthew) [reply apps/condo/domains/contact/components/CreateContactForm.tsx:38]: I used this translation before in https://github.com/open-condo-software/condo/pull/229 . But separated PRs for easier review...and I guess some lang changes are there...they will work after both PRs will be merged 👍

File: apps/condo/pages/contact/create.tsx (hunk 10 above)
Comment (SavelevMatthew) [reply apps/condo/pages/contact/create.tsx:18]: See comment below about missing few translation lines...

File: apps/condo/domains/contact/components/CreateContactForm.tsx (hunk 9 above)
Comment (SavelevMatthew) [reply apps/condo/domains/contact/components/CreateContactForm.tsx]: done 👍

File: apps/condo/domains/contact/components/CreateContactForm.tsx (hunk 8 above)
Comment (SavelevMatthew) [reply apps/condo/domains/contact/components/CreateContactForm.tsx]: done 👍

File: apps/condo/domains/contact/components/CreateContactForm.tsx (hunk 7 above)
Comment (SavelevMatthew) [reply apps/condo/domains/contact/components/CreateContactForm.tsx]: done 👍

File: apps/condo/domains/contact/components/CreateContactForm.tsx (hunk 11 above)
Comment (SavelevMatthew) [reply apps/condo/domains/contact/components/CreateContactForm.tsx:39]: You can check this by rebasing this branch on top of 495th

**Review by pahaz — APPROVED** [review]

**Review by AntonAL — APPROVED** [review]
